Assets/Scripts/Save System/SaveFileHandler.cs

```diff
@@ -0,0 +1,59 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json;
+using UnityEngine;
+
+namespace Cosmobot
+{
+    public class SaveFileHandler
+    {
+        private string save_file_path = "";
+        private string save_file_name = "";
+
+        public SaveFileHandler(string save_file_name)
+        {
+            this.save_file_name = save_file_name + ".json";
+            this.save_file_path = Path.Combine(Application.persistentDataPath, this.save_file_name);
+            Directory.CreateDirectory(Application.persistentDataPath);
+        }
+        public void Save(GameData game_data)
+        {
+            // For now we'll just overwrite the file each time
+            if (File.Exists(save_file_path))
+            {
+                Debug.Log("Save file already exists, deleting old one!");
+                File.Delete(save_file_path);
+            }
+            try
+            {
+                using FileStream stream = File.Create(save_file_path);
+                stream.Close();
+                File.WriteAllText(save_file_path, JsonConvert.SerializeObject(game_data));
+                Debug.Log("Save file path: " + save_file_path);
+            }
+            catch (System.Exception e)
+            {
+                Debug.LogError($"Failed to create save file due to:  + {e.Message} {e.StackTrace}");
+            }
```

Comment:
you can delete `if (File.Exists)` `Delete` and `Create`:

[File.WriteAllText](https://learn.microsoft.com/en-us/dotnet/api/system.io.file.writealltext?view=net-8.0)
> **Creates a new file**, writes the specified string to the file, and then closes the file. If the target file already exists, it is truncated and overwritten.